packages/common/src/extensions/slickRowBasedEdit.ts

```diff
@@ -0,0 +1,596 @@
+    const idProperty = this._grid.getOptions().datasetIdPropertyName ?? 'id';
+    const targetRow = this._editedRows.get(dataContext[idProperty]);
+    if (
+      (target.classList.contains(BTN_ACTION_DELETE) ||
+        target.parentElement?.classList.contains(BTN_ACTION_DELETE)) &&
+      this._gridService
+    ) {
+      if (this._addonOptions?.actionButtons?.deleteButtonPrompt) {
+        if (
+          !window.confirm(this._addonOptions.actionButtons.deleteButtonPrompt)
+        ) {
+          return;
+        }
+      }
+
+      this.toggleEditmode(dataContext, false);
+      this._gridService.deleteItem(dataContext);
+    } else if (
+      target.classList.contains(BTN_ACTION_EDIT) ||
+      target.parentElement?.classList.contains(BTN_ACTION_EDIT)
+    ) {
+      if (!this._addonOptions?.allowMultipleRows && this._editedRows.size > 0) {
+        return;
+      }
+
+      this.toggleEditmode(dataContext, true);
+    } else if (
+      target.classList.contains(BTN_ACTION_UPDATE) ||
+      target.parentElement?.classList.contains(BTN_ACTION_UPDATE)
+    ) {
+      if (
+        this._addonOptions?.actionButtons?.updateButtonPrompt &&
+        (targetRow?.editCommands.length || 0) > 0
+      ) {
+        if (
+          !window.confirm(this._addonOptions.actionButtons.updateButtonPrompt)
+        ) {
+          return;
+        }
+      }
+
+      if (this._addonOptions?.onBeforeRowUpdated) {
+        const result = await this._addonOptions.onBeforeRowUpdated(args);
+
+        if (result !== true) {
+          return;
+        }
+      }
+
+      this.removeUnsavedStylingFromRow(args.row);
+      this.toggleEditmode(dataContext, false);
+    } else if (
+      target.classList.contains(BTN_ACTION_CANCEL) ||
+      target.parentElement?.classList.contains(BTN_ACTION_CANCEL)
+    ) {
+      if (
+        this._addonOptions?.actionButtons?.cancelButtonPrompt &&
+        (targetRow?.editCommands.length || 0) > 0
+      ) {
+        if (
+          !window.confirm(this._addonOptions.actionButtons.cancelButtonPrompt)
+        ) {
+          return;
+        }
+      }
+
+      this.undoRowEdit(dataContext);
+      this.toggleEditmode(dataContext, false);
+    }
+  }
+
+  protected actionColumnFormatter(
+    _row: number,
+    _cell: number,
+    _value: any,
+    _columnDef: Column,
+    dataContext: any
+  ) {
+    const options = this._grid.getOptions();
+    const isInEditMode = this._editedRows.has(
+      dataContext?.[options.datasetIdPropertyName ?? 'id']
+    );
+
+    const actionFragment = document.createDocumentFragment();
+    actionFragment
+      .appendChild(
+        createDomElement('span', {
+          className:
+            `${
+              options.rowBasedEditOptions?.actionButtons?.editButtonClassName ||
+              'button-style padding-1px mr-2'
+            } action-btns ` + BTN_ACTION_EDIT,
+          title:
+            options.rowBasedEditOptions?.actionButtons?.editButtonTitle ||
```

Comment:
I also usually have a set of Locale translation keys for I18N, we could probably add them in the future though `editButtonTitleKey`, `deleteButtonTitleKey` and `updateButtonTitleKey`... (e.g.: `updateButtonTitleKey: 'UPDATE'`)

It's a lot of work to make everyone happy 😅 

but that might be ok to skip that since it's only for buttons?

Reply:
usually I'm the one being nitpicky about i18n ;) good catch